lib/rules/indent.js

```diff
@@ -722,23 +722,39 @@ module.exports = {
                 const parent = node.parent;
 
                 nodeIndent = getNodeIndent(parent).goodChar;
-                if (parentVarNode && parentVarNode.loc.start.line !== node.loc.start.line) {
+                if (!parentVarNode || parentVarNode.loc.start.line !== node.loc.start.line) {
```

Comment:
Previously, the `if` statement only applied if there was `VariableDeclarator` parent that started on a different line. For example, the first line of the following array should be indented:

```js
var foo =
    [
        1
    ];
```

But the following array should not be indented:

```js
var foo = [
    1
];
```

However, the result of this was that the rule would not add other necessary indentation (e.g. in `CallExpression` nodes) if it didn't have a `VariableDeclarator` parent. Since the `if` statement immediately below this one explicitly handles the `VariableDeclarator` case, it's not necessary for this check to be so strict.